entry-generation/godot-compiler-plugin/src/main/kotlin/godot/compilerplugin/CommonComponentRegistrar.kt

```diff
@@ -0,0 +1,59 @@
+package godot.compilerplugin
+
+import com.intellij.mock.MockProject
+import de.jensklingenberg.mpapt.common.MpAptProject
+import godot.annotation.processor.GodotAnnotationProcessor
+import org.jetbrains.kotlin.codegen.extensions.ClassBuilderInterceptorExtension
+import org.jetbrains.kotlin.compiler.plugin.*
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.extensions.StorageComponentContainerContributor
+import org.jetbrains.kotlin.js.translate.extensions.JsSyntheticTranslateExtension
+
+class CommonComponentRegistrar : ComponentRegistrar {
+    override fun registerProjectComponents(
+        project: MockProject,
+        configuration: CompilerConfiguration
+    ) {
+        val processor = GodotAnnotationProcessor(
+            configuration.get(CompilerPluginConst.CommandlineArguments.ENTRY_DIR_PATH)!!,
+            configuration.get(CompilerPluginConst.CommandlineArguments.GDNS_DIR_PATH)!!
+        )
+        val mpapt = MpAptProject(processor, configuration)
+        StorageComponentContainerContributor.registerExtension(project, mpapt)
+        ClassBuilderInterceptorExtension.registerExtension(project, mpapt)
+        JsSyntheticTranslateExtension.registerExtension(project, mpapt)
+    }
+}
+
+class CommonGodotKotlinCompilerPluginCommandLineProcessor : CommandLineProcessor {
+    companion object {
+        val GDNS_DIR_PATH_OPTION = CliOption(
+            CompilerPluginConst.CommandLineOptionNames.gdnsDirPathOption,
+            "<name>",
```

Comment:
what does `<name>` mean here? If it is used as a help text, maybe use a more descriptive text? i.e since this is for gdns dir, maybe `<path-to-gdns-dir>`.

Reply:
true! I changed it.